pykusto/_src/query.py

```diff
@@ -619,12 +624,28 @@ def by(self, *args: Union[_AssignmentToSingleColumn, BaseColumn, BaseExpression]
             self._by_assignments.append(_AssignmentToSingleColumn(_AnyTypeColumn(column_name), group_exp))
         return self
 
+    def shuffle_key(self, *keys: Union[_AssignmentToSingleColumn, BaseColumn, BaseExpression]):
+        if keys is not None:
+            for key in keys:
+                self._shuffle_keys.append(key)
+
+        self._shuffle_by_keys = True
+        return self
+
     def _compile(self) -> KQL:
-        result = f"summarize {', '.join(a.to_kql() for a in self._assignments)}"
+        result = f"{self._compile_summarize_with_shuffle_hint()} {', '.join(a.to_kql() for a in self._assignments)}"
         if len(self._by_assignments) != 0 or len(self._by_columns) != 0:
             result += f' by {", ".join(chain((c.kql for c in self._by_columns), (a.to_kql() for a in self._by_assignments)))}'
         return KQL(result)
 
+    def _compile_summarize_with_shuffle_hint(self) -> str:
+        if not self._shuffle_by_keys:
+            return "summarize"
+        if len(self._shuffle_keys) == 0:
```

Comment:
This choice seems peculiar to me. I think that passing an empty list is quite confusing. Why not add a boolean parameter to `shuffle_key()` that represents `hint.strategy=shuffle`?

Reply:
Not sure I understand but you mean.
`hint.strategy=shuffle` means - shuffle by all defined keys. it that case you don't want/need to pass any specific keys but you do want to still define the shuffle.
You can also shuffle by specific keys and not all of them - that is why the keys option exists.

You can find more information here
https://docs.microsoft.com/en-us/azure/data-explorer/kusto/query/shufflequery
(actually it is a concept worth being familiar with)

Reply:
Yes I understand (just read the docs before commenting)
What I meant is, that I want the pykusto to resemble KQL as much as possible.
so if `hint.strategy=shuffle` is desired, I suggest passing a boolean variable (e.g `hint_strategy_shuffle=True`), instead of not passing any keys. I understand it creates redundancy, but I feel it will be clearer this way.

Reply:
I don't understand.
Explicitly calling the `shuffle_key` method means you want to shuffle by keys. Why do we need another boolean value?